### ISSUE TITLE
Can't open Database created with MMex 1.3.2, 1.3.3 & 1.4.0 on Linux

### ISSUE THREAD

**Issue by MartinX3** [moneymanagerex/android-money-manager-ex#1117]
Android 7.1

Can't open Database created with MMex 1.3.2, 1.3.3 & 1.4.0 on Linux

error 14 unknown error

error
attempt to invoke virtual method android database cursor android database sqlite sqlitedatavase raw query (java.lang.string, java.lang.string[]) on a null object reference

tried to use function "optimize database"
data base got shrinked but still not usable in app

_the android beta app in the playstore can open my database_

**PS:
Main problem is, that app permissions are only checked once.
This problem was caused, because unknown reasons revoked the storage permission from this app at some point.**

**Comment by MisterY** [comment]
This is most probably related to the desktop version of MMEX. Please open an issue at the MoneyManagerEx repository. Thanks!

**Comment by MartinX3** [comment]
You closed that bug, without waiting for my reply? Realy?

And did you read _everything_ of my bug report?
I will quote an important part for you....
"the android beta app in the playstore can open my database"

And since April I'm using the 1.4 version of the desktop app.
Still, the normal old android app can't open the database anymore, while _the **beta** android app still can open it_.

You should update the standard android app with the improvements of the beta app in the database access code.

**Comment by MisterY** [comment]
Hi, @MartinX3. 
Please note that, to my understanding, a statement
> "the android beta app in the playstore can open my database"
does not indicate malfunction. Thank you for your further details. 
Currently I have no leads in terms of what exactly needs to be corrected. The database access code is identical between the two versions of the app.
If you can provide further information, that would be great. Also, please feel free to submit pull requests with fixes as there are currently no active Android developers on the project. Thanks!

**Comment by MartinX3** [comment]
Thank you very much for your fast response! :D
There are no active developers? Who make the updates? O.o

And I'm in the luck of beeing a programmer and java is one of the language, I'm coding in.
(Desktop, not mobile).
The only thing I miss is time.

The "master" branch is for the normal android app
and the "dev" branch is for the beta android app?

**Comment by MisterY** [comment]
:)
At the moment I seem to be the only semi-active contributor. I have the same issue as you, though. ;) After the app has been stabilised, I have moved to other areas and only occassionaly correct issues. The development is the easy and fun part but there is too much logistics involved in addition to development, for which I unfortunately do not have time.

You are correct in that the master is the stable branch and dev is the current development. Once the release is published, the code is merged from dev to master.
You can create a feature branch and submit a pull request. That would be the normal Git flow.
If you can debug and identify the source of the problem, that would be great. I can make small corrections, merge the code, and publish a Beta version to the community, should you identify any issues.
Please feel free to get yourself acquainted and contribute. :)
Thanks!

**Comment by MartinX3** [comment]
I tried the "master"-version again, but instead of choosing the database from the list of last used databased, i clicked on "diverses" (the german button to open the file explorer) and now the app asked me to get the permissions to access the storage.
I don't know why this app lost its permission, but now the app works again. :D

It should recheck the permissions, if you choose the database from the list, instead of the file explorer.
50% my fault (because I don't know why the permission got revoked) and 50% the fault of this app, which doesn't inform me about the missing permission :(

I will inform me, how I create a branch, to make a pull request, because I never coded with github. D:
I only code the company I work at the moment or for the university.

**Comment by MisterY** [comment]
Thanks for the info. Yes, the permissions are only checked once, after the app is installed. A whole lot of code would need to be modified in order to include asynchronous permissions check at various points in the app.
There are several ways that this could be handled in a user-friendly way. If you feel like it, please go ahead and implement something useful.

The good thing about the project is that the code is open and you can use it to learn while, at the same time, providing some benefits to everyone else involved.
I've spent several years actively coding this project, which I believe is enough from me. On the other hand, there are too many demanding users yet noone with conviction high enough to contribute some code modifications. ;)

**Comment by MartinX3** [comment]
I try to develop an useful implementation :)
Do I need to fork the project, to make a feature branch and make a pull request?
Because I don't find a "make branch" button :)

Year, open source is a great thing!
But only, if you read or update the code :P

**Comment by MisterY** [comment]
Great, thanks! :) It's a good opportunity to learn, if you are interested. You can read further on any topics or libraries involved and see what they do.

I would recommend you to start with a text like [this](http://blog.scottlowe.org/2015/01/27/using-fork-branch-git-workflow/). It is describing a general Git workflow. Indeed, you need to fork the original repository. Then also create a feature branch for the feature. Once implemented, the pull request you can submit to the MMEX dev branch from Github web site.
It is not difficult but requires some practice. You can't mess up. In the worst case, you just start again from the beginning.

Yes, an important issue is having to do everything alone, which is not fun at all. :) Well, the features not available elsewhere are also an important part of it, hence I try to maintain a working version of the app.
Welcome to OS development! :)

**Comment by MartinX3** [comment]
Thanks for the text :D
I try what I can to in my small amount of free time :D

And I hope you mean OSS development, instead of OS development.
OS (operating system) development would give me headaches. :P

I own IntelliJ and I saw, that it has GIT-Integration, maybe, there are the things built in, you are mentioning. :)

**Comment by MisterY** [comment]
A pull request will be created automatically as another issue. You can make manual references by adding # and an issue number.

**Comment by MartinX3** [comment]
I think since MMEX 1.5 uses the old DB version again, probably the app works again?

**Comment by wolfsolver** [comment]
Need to be closed since new 2023 version is aligned with desktop version 1.7.0
